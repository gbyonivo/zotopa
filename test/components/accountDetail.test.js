import React from 'react';
import { create } from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import combineReducers from '../../src/reducers';
import AccountDetails from '../../src/components/accountDetails';

const appState = {
  transactionReducer: {
    personalDetails: {
      3: {
        amount: 10, name: 'ivo', email: '3', date: 'd'
      }
    }
  }
};

describe('AccountDetails', () => {
  it('should render correctly with empty transactions', () => {
    const actual = create(
      <Provider store={createStore(combineReducers(appState))}>
        <AccountDetails personalDetails={{
          name: 'Ivo',
          email: 'io@io.com',
          availableFunds: '100000',
          transactions: []
        }} />
      </Provider>
    );
    expect(actual).toMatchSnapshot();
  });
  it('should render correctly with transactions', () => {
    const actual = create(
      <Provider store={createStore(combineReducers(appState))}>
        <AccountDetails personalDetails={{
          name: 'Ivo',
          email: 'io@io.com',
          availableFunds: '100000',
          transactions: [{
            amount: 10, name: 'ivo', email: '3', date: 'd'
          }]
        }} />
      </Provider>
    );
    expect(actual).toMatchSnapshot();
  });
});