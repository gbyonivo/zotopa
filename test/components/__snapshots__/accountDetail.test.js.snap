// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetails should render correctly with empty transactions 1`] = `
<ul
  className="sc-htpNat bWEXTj"
>
  <li>
    <label
      className="sc-bwzfXH EulNe"
    >
      Account Name
    </label>
    <span
      className="sc-bdVaJa dgblDn"
    >
      Ivo
    </span>
  </li>
  <li>
    <label
      className="sc-bwzfXH EulNe"
    >
      Account Number
    </label>
    <span
      className="sc-bdVaJa dgblDn"
    >
      io@io.com
    </span>
  </li>
  <li>
    <label
      className="sc-bwzfXH EulNe"
    >
      Availble Funds
    </label>
    <span
      className="sc-bdVaJa dgblDn"
    >
      £
      100000
    </span>
  </li>
  <li>
    <label
      className="transaction sc-bwzfXH EulNe"
    >
      Transactions
    </label>
    <span
      className="sc-bdVaJa dgblDn"
    >
      No Transactions
    </span>
  </li>
</ul>
`;

exports[`AccountDetails should render correctly with transactions 1`] = `
<ul
  className="sc-htpNat bWEXTj"
>
  <li>
    <label
      className="sc-bwzfXH EulNe"
    >
      Account Name
    </label>
    <span
      className="sc-bdVaJa dgblDn"
    >
      Ivo
    </span>
  </li>
  <li>
    <label
      className="sc-bwzfXH EulNe"
    >
      Account Number
    </label>
    <span
      className="sc-bdVaJa dgblDn"
    >
      io@io.com
    </span>
  </li>
  <li>
    <label
      className="sc-bwzfXH EulNe"
    >
      Availble Funds
    </label>
    <span
      className="sc-bdVaJa dgblDn"
    >
      £
      100000
    </span>
  </li>
  <li>
    <label
      className="transaction sc-bwzfXH EulNe"
    >
      Transactions
    </label>
    <table
      className=" responsiveTable"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th>
            Amount (total - £
            0
            )
          </th>
          <th>
            Date
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </li>
</ul>
`;
